simple_sso/sso_client/client.py

```diff
@@ -108,6 +108,10 @@ def get_user(self, access_token):
     def build_user(self, user_data):
         try:
             user = User.objects.get(username=user_data['username'])
+            # Update user data, excluding username changes
+            del user_data['username']
```

Comment:
This modifies the `user_data` dictionary that has been created by the caller and is not ideal.
A better approach is to check the keys in the for loop.

I'm also not sure about the security implications. What does user_data contain? Are all attributes meant to be reset?

Reply:
Good idea, keeping the original 'user_data'. I just pushed your requested change: https://github.com/divio/django-simple-sso/pull/62/commits/a0f864d973b6db9b238a86284847f185e8e854a7#diff-156d4d0f83e2028594d28cfe5b57a3aaec983a3722c45c6f9aaf0c7619a37c1dL112-R116 

Answering your question in here as well:
1. 'user_data' contains the serialized data from the sso server's 'User' object, extended with some extra_data, as seen in https://github.com/divio/django-simple-sso/blob/master/simple_sso/sso_server/server.py#L153-L166
2. All attributes beside the username are meant to be updated according to the user_data, which is provided on every user's login. This way the sso client will always have the updated user data from the sso server the next time the user logs into the sso client.
3. Regarding security implications: Of course one could add a set of 'editable_user_attrs' which defines the attributes, which can be updated on a login. But that is not needed when the project developers make sure no critical data is added to the 'extra_data' dict on the sso server's side, imho.